ui-components/src/helpers/filters.ts

```diff
@@ -1,70 +1,213 @@
+  getBackendFilterType() {
+    return this.name
+  }
+
+  getBackendFilterValue() {
+    return this.value
+  }
+}
+
+export class FrontEndFilters {
+  filters: Record<string, FrontEndFilter>
+
+  setValue(filterName: string, filterValue: any) {
+    this.filters[filterName].value = filterValue
+    if (filterName === COMMUNITY_TYPE && filterValue === ListingFilterKeys.seniorHousing) {
+      this.filters[ListingFilterKeys.seniorHousing].value = true
+    } else if (filterName === ListingFilterKeys.seniorHousing && filterValue == "true") {
+      this.filters[COMMUNITY_TYPE].value = ListingFilterKeys.seniorHousing
     }
   }
+
+  getFilterCount(): number {
+    let numberOfFilters = Object.keys(this.filters).filter(
+      (filterType) =>
+        this.filters[filterType].value !== undefined && this.filters[filterType].value != ""
+    ).length
+    // We want to consider rent as a single filter, so if both min and max are defined, reduce the count.
+    const hasMinMaxRentOverCount =
+      this.filters[ListingFilterKeys.minRent].value !== undefined &&
+      this.filters[ListingFilterKeys.maxRent].value != undefined
+    const hasSeniorHousingCommunityTypeOverCount =
+      this.filters[ListingFilterKeys.seniorHousing].value != undefined &&
+      this.filters[COMMUNITY_TYPE].value != undefined
+    if (hasMinMaxRentOverCount) {
+      numberOfFilters -= 1
+    }
+    if (hasSeniorHousingCommunityTypeOverCount) {
+      numberOfFilters -= 1
+    }
+    return numberOfFilters
+  }
+
+  constructor() {
+    this.filters = [
+      new CommunityTypeFilter(
+        COMMUNITY_TYPE,
+        EnumListingFilterParamsComparison["NA"],
+        communityTypeOptions
+      ),
+      new FrontEndFilter(
+        ListingFilterKeys.availability,
+        EnumListingFilterParamsComparison["NA"],
+        availabilityOptions
+      ),
+      new FrontEndFilter(
+        ListingFilterKeys.bedrooms,
+        EnumListingFilterParamsComparison[">="],
+        preferredUnitOptions
+      ),
+      new FrontEndFilter(ListingFilterKeys.zipcode, EnumListingFilterParamsComparison["IN"]),
+      new FrontEndFilter(ListingFilterKeys.minRent, EnumListingFilterParamsComparison[">="]),
+      new FrontEndFilter(ListingFilterKeys.maxRent, EnumListingFilterParamsComparison["<="]),
+      new FrontEndFilter(ListingFilterKeys.seniorHousing, EnumListingFilterParamsComparison["NA"]),
+      // Check if the filters below are used or should be deleted
+      new FrontEndFilter(ListingFilterKeys.name, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.leasingAgents, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.status, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.ami, EnumListingFilterParamsComparison["NA"]),
+    ].reduce(function (filters: Record<string, FrontEndFilter>, currFilter: FrontEndFilter) {
+      filters[currFilter.name] = currFilter
+      return filters
```

Comment:
The code right above this feels static; so I'm not sure what this reduce is doing?

Reply:
It's creating a mapping between the filterName and the FrontEndFilter so that other parts of the code can fetch filters by passing in the name.

Reply:
Thanks for explaining!

Reply:
I am not able to set it in the way you suggested, but I updated it to no longer use the reduce logic.